pkg/metrics/service-monitor.go

```diff
@@ -50,3 +83,12 @@ func GenerateServiceMonitor(s *v1.Service) *monitoringv1.ServiceMonitor {
 		},
 	}
 }
+
+// IsServiceMonitor checks if ServiceMonitor is registered in the cluster.
+func IsServiceMonitor(config *rest.Config) (bool, error) {
```

Comment:
Is there a use case for users to call this function directly? If not, maybe it should be unexported?  Same question applies to `GenerateServiceMonitor`.

Reply:
My idea behind this `GenerateServiceMonitor` was that the user doesn't want to create the resources directly sometimes just wants to generate the ServiceMonitor and create them later. Also, we already have this function in the previous releases and didn't want to break anyone already using it. As for the `IsServiceMonitor` I guess we can unexport it.